### ISSUE TITLE
Examples use mis-mash of 0.6 and 0.7 import paths

### ISSUE THREAD

**Issue by krehermann** [CQCL/pytket#42]
It seems the v0.6 imports were like
pytket.backends.
pytket.qiskit.

and in 0.7
pytket.extensions.backends.
pytket.extensions.qiskit

Some example like use 0.6 paths (e.g. https://github.com/CQCL/pytket/blob/master/examples/backends_example.ipynb) and some use 0.7 paths (e.g https://github.com/CQCL/pytket/blob/master/examples/circuit_analysis_example.ipynb, which very confusing.

**Comment by ss2165** [comment]
pytket 0.7 moved all the extensions to the `pytket.extensions` namespace. In order to not break too much code by doing this, we put in some redirects so that `pytket.backends.ibm` etc, still works (even though the true path is `pytket.extensions.backends.ibm`) as backend imports were the most common use of extensions by far. The non-backend funcitonality in some extensions (e.g. `tk_to_qiskit`) now come only from `pytket.extensions.qiskit`. This is the distinction you see in the examples, for the backends we are using the old backends path for simplicity, but for other tools we have changed the import. I agree this is a bit confusing, we may deprecate the old paths and strictly require the new ones in future.

**Comment by krehermann** [comment]
If i undertand correctly,

pytket.backends.ibm 
is intended to work in 0.71. Is this correct?

It was not the case for me. This is the error I received,

    from pytket.backends.ibm import AerStateBackend, AerBackend
ModuleNotFoundError: No module named 'pytket.backends.ibm'

which caused me to go digging the in the RECORD file of the pip install to figure out that I needed
from pytket.extensions.backends.ibm import AerStateBackend, AerBackend

which does work for me in 0.71

**Comment by ss2165** [comment]
Thank you for the clarification, that does look like an issue. We have not been able to replicate it on some systems we've tried, could you let me know which OS version and python environment you were using?

**Comment by krehermann** [comment]
Sure,

Summary:
```
python: 3.7.9
os: Ubuntu 18.04
pytket: 0.7.1.1
```

details:
Full error:
```
/home/krehermann/.virtualenvs/qc-3.7/bin/python /home/krehermann/git/oreilly-qc.github.io/samples/Pytket/ch03_02_entangled_qubits.py
Traceback (most recent call last):
  File "/home/krehermann/git/oreilly-qc.github.io/samples/Pytket/ch03_02_entangled_qubits.py", line 2, in <module>
    from pytket.backends.ibm import AerStateBackend, AerBackend
ModuleNotFoundError: No module named 'pytket.backends.ibm'

$ /home/krehermann/.virtualenvs/qc-3.7/bin/python --version
Python 3.7.9
$ /home/krehermann/.virtualenvs/qc-3.7/bin/pip
pip     pip3    pip3.7  
$ /home/krehermann/.virtualenvs/qc-3.7/bin/pip show pytket
Name: pytket
Version: 0.7.1.1
Summary: Python module for interfacing with the CQC tket library of quantum software
Home-page: https://cqcl.github.io/pytket
Author: Seyon Sivarajah
Author-email: seyon.sivarajah@cambridgequantum.com
License: CQC Software Licence
Location: /home/krehermann/.virtualenvs/qc-3.7/lib/python3.7/site-packages
Requires: numpy, networkx, scipy, sympy, lark-parser, graphviz
Required-by: pytket-qulacs, pytket-qiskit, pytket-pyzx, pytket-cirq
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.5 LTS
Release:        18.04
Codename:       bionic
```

**Comment by ss2165** [comment]
I have been able to replicate this now, thank you. 

The issue seems to be with installing the latest versions of extensions as upgrades, if the old namespace structure is already present in the environment then the new structure does not get installed correctly. We will look in to avoiding this with future releases. 

In the meantime, uninstallting pytket before  updating seems to cause correct behaviour.

For example by running

`pip install --upgrade --force-reinstall pytket pytket-qiskit`

**Comment by lmondada** [comment]
The issue was caused by filename conflicts that appeared when upgrading `pytket` from 0.6 to 0.7. I am closing this issue as this is solved by the latest 0.8 release.
